### ISSUE TITLE
[Subtask] [spark-connector] support querying with Hive function and UDF

### ISSUE THREAD

**Issue by FANNG1** [apache/gravitino#1566]
### Describe the subtask

support querying with Hive function and UDF

### Parent issue

#1227

**Comment by yangyuxia** [comment]
When use Spark connector for cross-domain federated query, we found that we cannot use user-defined functions defined in Hive: Error in query: Catalog hive_catalog does not support functions.
hive udf:

![Image](https://github.com/user-attachments/assets/ddbb91ed-9773-48a7-a02d-cd780096137e)

![Image](https://github.com/user-attachments/assets/2c5b48df-4fc5-409e-950d-1c44fcb98564)

spark use udf:

![Image](https://github.com/user-attachments/assets/e928321b-71e5-4e73-b624-d3242e748257)

**Comment by FANNG1** [comment]
Yes, this is one of the limitations of kyuubi spark connector which doesn's support `FunctionCatalog` yet.
